src/System.IO/tests/MemoryStream/MemoryStream.GetBufferTests.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.IO.Tests
+{
+    public class MemoryStream_GetBufferTests
+    {
+        [Fact]
+        public void MemoryStream_GetBuffer_Length()
+        {
+            MemoryStream ms = new MemoryStream();
+            byte[] buffer = ms.GetBuffer();
+            Assert.Equal(0, buffer.Length);
+        }
+
+        [Fact]
+        public void MemoryStream_GetBuffer_NonExposable()
+        {
+            MemoryStream ms = new MemoryStream(new byte[100]);
+            Assert.Throws<UnauthorizedAccessException>(() => ms.GetBuffer());
```

Comment:
I may have missed it, but it'd be good to add a test using one of MemoryStream's constructors that accepts a byte array but also allows it to be exposed.